### PR TITLE
[Fixes #332] Update menu items based on the user permissions

### LINKED ISSUE #332: Update menu items based on the user permissions
We moved the navbars from react to django templates in recent updates so now we need to provide the correct list of items using the templatetags. There are currently two functions that provides static list of items:

- [get_user_menu](https://github.com/GeoNode/geonode-mapstore-client/blob/master/geonode_mapstore_client/templatetags/get_menu_json.py#L145) get the menu items for the top right user menus

- [get_base_left_topbar_menu](https://github.com/GeoNode/geonode-mapstore-client/blob/master/geonode_mapstore_client/templatetags/get_menu_json.py#L16) get the menu items for blue topbar

These are the expected lists:

Note: there are some comments inline in the item objects

### get_user_menu

- not authenticated user

```py
{
    "label": "Register",
    "type": "link",
    "href": "/account/signup/?next=/" # if possible next should point to the current location
},
{
    "label": "Sign in",
    "type": "link",
    "href": "/account/login/?next=/" # if possible next should point to the current location
}
```

- authenticated user (see notes for role admin)
```py
{
    "image": "path/to/avatar", # get src of user avatar
    "type": "dropdown",
    "className": "gn-user-menu-dropdown",
    "items": [
        {
            "type": "link",
            "href": "path/to/user/profile", # get href of user profile
            "label": "Profile"
        },
        {
            "type": "link",
            "href": "/social/recent-activity",
            "label": "Recent activity"
        },
        {
            "type": "link",
            "href": "/favourite/list/",
            "label": "Favorites"
        },
        {
            "type": "link",
            "href": "/messages/inbox/",
            "label": "Inbox"
        },
        {
            "type": "divider"
        },
        { # add this item only if the user has admin role
            "type": "link",
            "href": "/admin/",
            "label": "Admin"
        },
        { # add this item only if the user has admin role and geoserver app is available
            "type": "link",
            "href": "/geoserver/",
            "label": "GeoServer"
        },
        { # add this item only if the user has admin role
            "type": "divider"
        },
        {
            "type": "link",
            "href": "/help/",
            "label": "Help"
        },
        {
            "type": "divider"
        },
        {
            "type": "link",
            "href": "/account/logout/?next=/", # if possible next should point to the current location
            "label": "Log out"
        }
    ]
}
```

### get_base_left_topbar_menu 

- not authenticated user 
```py
{
    "label": "Datasets",
    "type": "dropdown",
    "items": [
        {
            "type": "link",
            "href": "/catalogue/#/search/?filter{resource_type.in}=layer",
            "label": "Datasets",
            "badge": 0 # total count of datasets or undefined
        }
    ]
},
{
    "label": "Documents",
    "type": "dropdown",
    "items": [
        {
            "type": "link",
            "href": "/catalogue/#/search/?filter{resource_type.in}=document",
            "label": "Documents",
            "badge": 0 # total count of documents or undefined
        }
    ]
},
{
    "type": "divider"
},
{
    "label": "Maps",
    "type": "dropdown",
    "items": [
        {
            "type": "link",
            "href": "/catalogue/#/search/?filter{resource_type.in}=map",
            "label": "Explore maps",
            "badge": 0 # total count of maps or undefined
        }
    ]
},
{
    "label": "GeoStories",
    "type": "dropdown",
    "items": [
        {
            "type": "link",
            "href": "/catalogue/#/search/?filter{resource_type.in}=geostory",
            "label": "GeoStories",
            "badge": 0 # total count of geostories or undefined
        }
    ]
},
{
    "label": "About",
    "type": "dropdown",
    "items": [
        {
            "type": "link",
            "href": "/people/",
            "label": "People"
        },
        {
            "type": "link",
            "href": "/groups/",
            "label": "Groups"
        },
        {
            "type": "link",
            "href": "/groups/categories/",
            "label": "Groups categories"
        }
    ]
}
```

- authenticated user with `add_resource` permission (see notes for role admin)

```py
{
    "label": "Datasets",
    "type": "dropdown",
    "items": [
        {
            "type": "link",
            "href": "/catalogue/#/search/?filter{resource_type.in}=layer",
            "label": "Datasets",
            "badge": 0 # total count of datasets or undefined
        },
        {
            "type": "divider"
        },
        {
            "type": "link",
            "href": "/layers/upload",
            "label": "Upload dataset"
        }
    ]
},
{
    "label": "Documents",
    "type": "dropdown",
    "items": [
        {
            "type": "link",
            "href": "/catalogue/#/search/?filter{resource_type.in}=document",
            "label": "Documents",
            "badge": 0 # total count of documents or undefined
        },
        {
            "type": "divider"
        },
        {
            "type": "link",
            "href": "/documents/upload",
            "label": "Upload document"
        }
    ]
},
{
    "type": "divider"
},
{
    "label": "Maps",
    "type": "dropdown",
    "items": [
        {
            "type": "link",
            "href": "/catalogue/#/search/?filter{resource_type.in}=map",
            "label": "Explore maps",
            "badge": 0 # total count of maps or undefined
        },
        {
            "type": "divider"
        },
        {
            "type": "link",
            "href": "/catalogue/#/map/new",
            "label": "Create Map"
        }
    ]
},
{
    "label": "GeoStories",
    "type": "dropdown",
    "items": [
        {
            "type": "link",
            "href": "/catalogue/#/search/?filter{resource_type.in}=geostory",
            "label": "GeoStories",
            "badge": 0 # total count of geostories or undefined
        },
        {
            "type": "divider"
        },
        {
            "type": "link",
            "href": "/catalogue/#/geostory/new",
            "label": "Create GeoStory"
        }
    ]
},
{
    "label": "About",
    "type": "dropdown",
    "items": [
        {
            "type": "link",
            "href": "/people/",
            "label": "People"
        },
        {
            "type": "link",
            "href": "/groups/",
            "label": "Groups"
        },
        {
            "type": "link",
            "href": "/groups/categories/",
            "label": "Groups categories"
        },
        { # add this item only if the user has admin role
            "type": "link",
            "href": "/announcements/",
            "label": "Announcements"
        },
        {
            "type": "divider"
        },
        {
            "type": "link",
            "href": "/invitations/geonode-send-invite/",
            "label": "Invite users"
        },
        { # add this item only if the user has admin role
            "type": "link",
            "href": "/admin/people/profile/add/",
            "label": "Add user"
        },
        { # add this item only if the user has admin role
            "type": "link",
            "href": "/groups/create/",
            "label": "Create group"
        }
    ]
}
```

### PR DESCRIPTION
**Pull request by marthamareal**

### CODE REVIEW

File: geonode_mapstore_client/templatetags/get_menu_json.py [hunk 1]
```diff
@@ -125,109 +133,123 @@ def get_base_right_topbar_menu():
...
+def get_user_menu(context):
+    user = context.get('request').user
+
+    if not user.is_authenticated:
+        return [
+            {
+                "label": "Register",
+                "type": "link",
+                "href": "/account/signup/?next=/"
+            } if settings.ACCOUNT_OPEN_SIGNUP else None,
```
Comment (giohappy) [inline geonode_mapstore_client/templatetags/get_menu_json.py:172]: missing read-only check @marthamareal

**Review by giohappy — CHANGES_REQUESTED** [review]

File: geonode_mapstore_client/templatetags/get_menu_json.py [hunk 2]
```diff
@@ -12,96 +16,100 @@ def _handle_single_item(menu_item):
...
+        "items": [
+            {
+                "type": "link",
+                "href": "/catalogue/#/search/?filter{resource_type.in}=geostory",
+                "label": "GeoStories"
+            }
+        ]
+    }
+    user = context.get('request').user
+    if user.is_authenticated:
```
Comment (allyoucanmap) [inline geonode_mapstore_client/templatetags/get_menu_json.py:71]: just a note all the following link should be enabled only if user has add resource permissions:

- Upload dataset
- Upload document
- Create map
- Create GeoStory

File: geonode_mapstore_client/templatetags/get_menu_json.py [hunk 3]
```diff
@@ -125,109 +133,123 @@ def get_base_right_topbar_menu():
...
+                "href": "/account/login/?next=/"
+            },
+        ]
+
+    devider = {
+        "type": "divider"
+    }
+    profile = {
+        # get src of user avatar
+        "image": "https://www.gravatar.com/avatar/7a68c67c8d409ff07e42aa5d5ab7b765/?s=240",
```
Comment (allyoucanmap) [inline geonode_mapstore_client/templatetags/get_menu_json.py:185]: here we should add the actual avatar of the user

File: geonode_mapstore_client/templatetags/get_menu_json.py [hunk 4]
```diff
@@ -125,109 +133,123 @@ def get_base_right_topbar_menu():
...
+        "type": "dropdown",
+        "className": "gn-user-menu-dropdown",
+        "items": [
+            {
+                "type": "link",
+                # get href of user profile
+                "href": "{state('user') && state('user').hrefProfile}",
```
Comment (allyoucanmap) [inline geonode_mapstore_client/templatetags/get_menu_json.py:192]: here is missing the href for the user profile

**Review by allyoucanmap — CHANGES_REQUESTED** [review]

File: geonode_mapstore_client/templatetags/get_menu_json.py (hunk 2 above)
Comment (allyoucanmap) [reply geonode_mapstore_client/templatetags/get_menu_json.py:71]: Missing also badge property with the total count of resources by type

**Comment by marthamareal** [comment]
updated

**Review by allyoucanmap — APPROVED** [review]

**Review by giohappy — APPROVED** [review]

**Pull request merged**